moveit_core/CMakeLists.txt

```diff
@@ -7,6 +7,8 @@ endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
There's been some discussion about using this in ROS 2. MoveIt2 related discussion/comments: https://github.com/ros-planning/moveit2/pull/286.

Is this something we want to diverge on here in `moveit`?

Reply:
I guess I haven't realised how big of a task it would be to fix all of moveit. Shall I close this PR and open an issue instead? I don't plan to continue working on this.

Reply:
As there are already approaches available to compile MoveIt on Windows, I think this PR could be closed and instead, an issue should be opened to track and join all those other activities.